Automattic/woocommerce-payments
# Redirect fo the "Connect" page on plugin activation or when trying to navigate to the settings screen

In this PR:
- On any admin screen, if WCPay is not setup, the merchant will be immediately redirected to the WCPay Connect screen.
- Once that happens, a flag is set in `wp_options` so it doesn't happen again. So, realistically, it will just happen the first time the plugin is installed or when the plugin is updated from an old version to a version that includes this PR.
- Because of that redirect, [the admin notice](https://user-images.githubusercontent.com/1715800/80979933-e37d6e00-8e1f-11ea-8d65-792454e7e119.png) nudging the merchant to connect is no longer needed. So I removed it.
- If the merchant goes to the `WooCommerce -> Settings -> Payments -> WCPay` screen and WCPay is not setup, he will be immediately redirected to the Connect page. That will change the breadcrumbs but I think it's an OK compromise because react-ifying the settings screen while [keeping the tab bar](https://user-images.githubusercontent.com/1715800/80979617-7d90e680-8e1f-11ea-9d7a-27a694f9a35b.png) may be tricky or impossible.

To test:
- Make sure you don't have your WCPay account connected.
- Update the plugin to this PR, and open any WP-Admin page.
- You will be immediately redirected to the Connect screen.
- Keep navigating on WP-Admin, notice you won't be redirected again.
- Go to `WooCommerce -> Settings -> Payments -> WCPay`, you'll be redirected to the Connect screen.

cc/ @LevinMedia Are you OK with this behaviour?
